operator/internal/migrations/storageschema/storageschema_configmap.go

```diff
@@ -0,0 +1,90 @@
+package storageschema
+
+import (
+	"context"
+	"fmt"
+
+	lokiv1beta1 "github.com/grafana/loki/operator/api/v1beta1"
+	"github.com/grafana/loki/operator/internal/external/k8s"
+	"github.com/grafana/loki/operator/internal/manifests"
+	"github.com/grafana/loki/operator/internal/manifests/storage"
+
+	"github.com/ViaQ/logerr/kverrors"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/json"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+)
+
+type lokiConfigJSON struct {
+	Schema lokiSchemaConfigJSON `json:"schema_config"`
+}
+
+type lokiSchemaConfigJSON struct {
+	Configs []lokiStorageSchemaJSON `json:"configs"`
+}
+
+type lokiStorageSchemaJSON struct {
+	From    string `json:"from"`
+	Version string `json:"schema"`
+}
```

Comment:
nit. What is the purpose of the `JSON` suffix here? These types are neither exported nor used anywhere else. The should be named after the config sections that pass through unmarshaling.